gsd/fl.pyx

```diff
@@ -848,6 +842,24 @@ cdef class GSDFile:
 
         return retval
 
+    def upgrade(self):
+        """ upgrade()
+
+        Upgrade a GSD file to the v2 specification in place. The file must be
+        open in a writable mode.
+
+        """
+
+        if not self.__is_open:
+            raise ValueError("File is not open")
+
+        logger.info('upgrading file: ' + self.name)
+
+        with nogil:
+            retval = libgsd.gsd_upgrade(&self.__handle)
```

Comment:
GSD itself is not multi-threaded. I use `with nogil` throughout GSD when it drops into a pure C call so that multithreaded user scripts do not serialize on GSD (e.g. one thread for reading data from GSD, one thread for processing that data).

Reply:
Does that primarily impact usage of the C API? Is it possible to accomplish something like that from pure Python, or are you considering Cython usage of GSD?

Reply:
Discussed offline: releasing the GIL here potentially allows multiple threads to be reading GSD files simultaneously, or to use some sort of queue to potentially consume chunks of the GSD file (or multiple GSD files) being added to a queue in separate threads from where they're written.